client/blocks/props-viewer/index.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * External Dependencies
+ */
+import React, { PropTypes, PureComponent } from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+
+// todo: move this file!
+const components = require( '../../../server/devdocs/proptypes-index.json' );
+
+/**
+ * Finds a non-example component in the library of components
+ * @param {string} slug The slug to search for
+ * @return {Array} An array of component matches
+ */
+export function findRealComponent( slug ) {
+	// remove the last character. As of right now, all plural display names are with just an 's'
+	const singular = slug.slice( 0, -1 );
+	return components.filter( ( component ) => {
+		return ( slug === component.slug || singular === component.slug ) && component.includePath.indexOf( 'example' ) < 0;
+	} );
+}
+
+/**
+ * Renders a table of prop-types for auto-documentation
+ */
+class PropsViewer extends PureComponent {
+	static displayName = 'PropsViewer';
+
+	static propTypes = {
+		/**
+		 * The slug of the component being displayed
+		 */
+		component: PropTypes.string.isRequired,
+
+		/**
+		 * The element to display as an example of this component
+		 */
+		example: PropTypes.element.isRequired
+	}
+
+	constructor( props ) {
+		super( props );
+	}
+
+	/**
+	 * Set the state of this component to the first matching slug
+	 * @param {string} slug The slug to search for
+	 */
+	setComponent = ( slug ) => {
+		let component = findRealComponent( slug );
+		if ( component.length > 0 ) {
+			component = component[ 0 ];
+		} else {
+			component = null;
+		}
+
+		this.setState( {
+			component
+		} );
+	};
+
+	componentWillMount() {
```

Comment:
Looks like it's been removed from the official docs, but this seems close to falling within the anti-pattern of using props in state. Here's an archive of the original text:

https://github.com/vjeux/react-website/blob/master/src/react/content/tips/10-props-in-getInitialState-as-anti-pattern.md

tl;dr: It becomes ugly quick to try to keep state in sync in response to props, when you may just as well component result of `setComponent` at time of render (it's probably not as slow as you might think it to be).